### ISSUE TITLE
Problems on windows

### ISSUE THREAD

**Issue by temp** [puli/issues#102]
* A puli.BAT wrapper is installed in bin on Windows systems. The PuliRunner has an exception for this, and removes the prefix php binary, but checks for ".bat", not ".BAT".
* PuliPlugin::loadPuliPackages() calls "package --list --format %name%,..." on the puliRunner, but this produces a non-parsable output on windows systems.
Expected is
```
webmozart/assert;composer;C:/Users/pfahsel/projects/tipfinder/vendor/webmozart/assert;enabled
...
```
but the output is
```
'webmozart/assert;composer;C:/Users/pfahsel/projects/tipfinder/vendor/webmozart/assert;enabled'
```
* PuliPlugin::installNewPackages() ALWAYS reinstalls all packages on Windows. This is because the check $installPath === $puliPackages[$packageName]->getInstallPath() differ, the left hand side uses partly backslash directory delimiters, the right hand side uses slash directory delimiters. Because of this every install path differs, and every package is always reinstalled.
* Puli doesn't find any resources.

Points 3+4 are because of inconsistencies between the repository paths and dynamically generated paths, which are compared. One uses mixed backslashes and slashes in the path, the other only slashes. This leads to failing comparisons.

**Comment by webmozart** [comment]
Thanks for the report! Do you want to work on a fix?

**Comment by temp** [comment]
Point 1+2 are fairly easy, yes. But point 3 goes too deep, I don't know where all the path mappings are done.

**Comment by temp** [comment]
Which leads to the next points, after doing a quick fix for the above problems, it showed that puli doesn't work at all on windows, with the current version.
We tried to debug, but it's somewhere deep in the path mappings.
In the repo slashes and backslashes are mixed, like "C:/path/to/resources\", and after that all comparisons break. We tried to fix this, but failed.

**Comment by temp** [comment]
Reproduce:

1. mkdir res
2. touch res/test.txt
3. puli.phar map res res
4. puli.phar find --name *.txt

Result:
empty ;-)

**Comment by temp** [comment]
Same with the prior version of puli:

```
$ ./puli-old.phar find --name *.txt
FileResource /res/bla.txt
$ ./puli.phar find --name *.txt
<empty>
```

**Comment by webmozart** [comment]
Which version are you using?

**Comment by temp** [comment]
```
$ ./puli.phar --version
Puli version 1.0.0-beta6
$ ./puli-old.phar --version
Puli version 1.0.0-beta5
```

**Comment by temp** [comment]
On my mac, everything works fine. Here at my colleagues windows machine, everything breaks ;-)

**Comment by temp** [comment]
Point 3 was fixed by a commit from @webmozart

**Comment by temp** [comment]
@webmozart Point 4 is also fixed! It was the simply the first Path::normalize() we added to PathMappingRepository::getFilesystemResourceChildren().
But I guess other repositories also have the problem...

**Comment by webmozart** [comment]
The FilesystemRepository tests were fixed on Windows in puli/repository#44. A few PathMappingRepository tests remain broken, but @tgalopin is working on that. :)

**Comment by webmozart** [comment]
The last Windows issues are fixed in puli/repository#46.
